src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -20,27 +24,50 @@ public abstract class OwnedBuffer<T> : IDisposable, IKnown
 
         public virtual BufferHandle Pin(int index = 0)
```

Comment:
method signature unsafe is simply a shortcut for unsafe block in the implementation, i.e. it's a property of implementation despite the fact that it appears in the signature.